### PR TITLE
Replace deprecated validate_* functions

### PR DESCRIPTION
**Pull request by baurmatt**

### CODE REVIEW

File: manifests/global_hook.pp [hunk 1]
```diff
@@ -77,10 +74,6 @@
     fail('gitlab::custom_hook resource must specify either content or source, but not both')
   }
 
-  if $source {
-    validate_re($source, '^puppet:')
-  }
-
   $hook_path = "${_custom_hooks_dir}/${type}.d"
```
Comment (Dan33l) [inline manifests/global_hook.pp]: What does it means with `undef` as default value for `$type` and `$custom_hooks_dir` ?

It looks better to not provide defaults at all and remove `Optional` for parameters `$type` and `$custom_hooks_dir`.

**Comment by Dan33l** [comment]
Hi @baurmatt thank you for this PR.
I added one inline comment.

**Comment by baurmatt** [comment]
@Dan33l Thanks! :) Did you forget to save it? I can't see it :(

File: manifests/global_hook.pp [hunk 2]
```diff
@@ -52,22 +52,19 @@
...
 # Copyright 2017 Hidde Boomsma
 #
-define gitlab::global_hook(
-  $type = undef,
-  $content = undef,
-  $source = undef,
-  $custom_hooks_dir = undef
+define gitlab::global_hook (
+  Pattern[/^(post-receive|pre-receive|update)$/] $type,
+  Optional[Stdlib::Absolutepath]                 $custom_hooks_dir,
```
Comment (Dan33l) [inline manifests/global_hook.pp]: `Optional` datatype yet present ?

File: manifests/global_hook.pp (hunk 1 above)
Comment (Dan33l) [reply manifests/global_hook.pp]: `Optional` datatype yet present with `$custom_hooks_dir` ?

File: manifests/global_hook.pp (hunk 2 above)
Comment (baurmatt) [reply manifests/global_hook.pp]: Dammit :D Fixed! :)

File: manifests/global_hook.pp [hunk 3]
```diff
@@ -52,22 +52,19 @@
 #
 # Copyright 2017 Hidde Boomsma
 #
-define gitlab::global_hook(
-  $type = undef,
-  $content = undef,
-  $source = undef,
-  $custom_hooks_dir = undef
+define gitlab::global_hook (
+  Pattern[/^(post-receive|pre-receive|update)$/] $type,
```
Comment (alexjfisher) [inline manifests/global_hook.pp]: `Enum` is a better choice here?

Comment (alexjfisher) [reply manifests/global_hook.pp]: `Enum['post-receive', 'pre-receive', 'update'] $type`

Comment (baurmatt) [reply manifests/global_hook.pp]: Sure :)

**Pull request merged**